testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -624,7 +624,7 @@ void testCastExactNumericLimits(CastType castType, SqlOperatorFixture f) {
         f.checkCastFails(numeric.minOverflowNumericString,
             type, LITERAL_OUT_OF_RANGE_MESSAGE, false, castType);
       } else {
-        if (Bug.CALCITE_2539_FIXED) {
+        if (numeric != Numeric.DECIMAL5_2 || Bug.CALCITE_2539_FIXED) {
           f.checkCastFails(numeric.maxOverflowNumericString,
               type, OUT_OF_RANGE_MESSAGE, true, castType);
```

Comment:
It seems that `OUT_OF_RANGE_MESSAGE` is defined as a "generic string"; in the tests that you have re-enabled, could we replace it for a more meaningful value with the actual expected error message?

Reply:
Yes, sorry about that; unfortunately the error message produced by ParseException is not very good.

Reply:
Thanks for the improvements @mihaibudiu . One last question, shouldn't we adjust the expected error message in these lines for the actual expected error message (instead of `OUT_OF_RANGE_MESSAGE`)?
```
 if (numeric != Numeric.DECIMAL5_2 || Bug.CALCITE_2539_FIXED) {
  f.checkCastFails(numeric.maxOverflowNumericString,
    type, OUT_OF_RANGE_MESSAGE, true, castType);  // <-- *****
  f.checkCastFails(numeric.minOverflowNumericString,
    type, OUT_OF_RANGE_MESSAGE, true, castType);  // <-- *****
}
```

Reply:
I think the test does not fail because `OUT_OF_RANGE_MESSAGE` is currently defined as `(?s).*`, so any error message would be fine (and I think we could be a bit more specific and test an actually expected error thanks to your improvements).

Reply:
This depended on a BUG_* which I couldn't locate anymore, so I improved the message and removed the reference to the BUG altogether.